pandas/core/indexes/api.py

```diff
@@ -298,15 +298,16 @@ def all_indexes_same(indexes):
 
     Parameters
     ----------
-    indexes : list of Index objects
+    indexes : iterable of Index objects
 
     Returns
     -------
     bool
         True if all indexes contain the same elements, False otherwise.
     """
-    first = indexes[0]
-    for index in indexes[1:]:
+    itr = iter(indexes)
```

Comment:
Is this change related to the PR?

Reply:
Yes - it allows for this line:

    all_indexed_same = all_indexes_same((x.index for x in values))

Previously a full list was created and then passed to `all_indexes_same`. By using a generator, we allow the possibility to short-circuit if say the first to indices are different.